go/cache/lru_cache_test.go

```diff
@@ -121,31 +108,29 @@ func TestCapacityIsObeyed(t *testing.T) {
 	cache.Set("key1", value)
 	cache.Set("key2", value)
 	cache.Set("key3", value)
-	if sz := cache.UsedCapacity(); sz != size {
-		t.Errorf("cache.UsedCapacity() = %v, expected %v", sz, size)
-	}
+	sz := cache.UsedCapacity()
+	assert.Equal(t, size, sz, "cache.UsedCapacity() = %v, expected %v", sz, size)
 	// Insert one more; something should be evicted to make room.
 	cache.Set("key4", value)
 	sz, evictions := cache.UsedCapacity(), cache.Evictions()
 	assert.Equal(t, size, sz)
 	assert.EqualValues(t, 1, evictions)
 
 	// Check various other stats
-	if l := cache.Len(); int64(l) != size {
-		t.Errorf("cache.Len() returned bad length: %v", l)
-	}
-	if s := cache.UsedCapacity(); s != size {
-		t.Errorf("cache.UsedCapacity() returned bad size: %v", s)
-	}
-	if c := cache.MaxCapacity(); c != size {
-		t.Errorf("cache.UsedCapacity() returned bad length: %v", c)
-	}
-	if c := cache.Hits(); c != 0 {
-		t.Errorf("cache.Hits() returned hits when there should be none: %v", c)
-	}
-	if c := cache.Misses(); c != 0 {
-		t.Errorf("cache.Misses() returned misses when there should be none: %v", c)
-	}
+	l := cache.Len()
+	assert.Equal(t, size, int64(l), "cache.Len() returned bad length: %v", l)
```

Comment:
Let's use `EqualValues` which accepts compatible integer types.

```suggestion
	EqualValues(t, size, l, "cache.Len() returned bad length: %v", l)
```